shipping/uc_usps/uc_usps.admin.inc

```diff
@@ -24,77 +24,203 @@ function uc_usps_admin_settings($form, &$form_state) {
   );
 
+  $form['domestic']['newapi']['uc_usps_newapi_dom_parcel_services'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('USPS small package services'),
+    '#options' => $config->get('uc_usps_newapi_dom_parcel_all_services'),
+    '#default_value' => $config->get('uc_usps_newapi_dom_parcel_services'),
+    '#description' => t('Select the USPS services that are available to customers. Be sure to include the services that the Postal Service agrees are available to you.'),
+  );
+
+  // Container for international quotes
   $form['international'] = array(
     '#type' => 'fieldset',
-    '#title' => t('USPS International'),
-    '#description' => t('Set the conditions that will return a USPS International quote.'),
+    '#title' => t('USPS international'),
+    '#description' => t('Set the conditions that will return a USPS international quote.'),
     '#collapsible' => TRUE,
     '#collapsed' => TRUE,
     '#group' => 'usps-settings',
   );
-
-  $form['international']['uc_usps_intl_env_services'] = array(
+  $form['international']['legacy'] = array(
+    '#type' => 'fieldset',
+    '#states' => array(
+      'visible' => array(
+        ':input[name="uc_usps_authorization_method"]' => array('value' => 'legacy'),
+      ),
+    ),
+  );
+  $form['international']['legacy']['uc_usps_intl_env_services'] = array(
     '#type' => 'checkboxes',
     '#title' => t('USPS international envelope services'),
-    '#default_value' => $config->get('uc_usps_intl_env_services'),
     '#options' => _uc_usps_intl_env_services(),
+    '#default_value' => $config->get('uc_usps_intl_env_services'),
     '#description' => t('Select the USPS services that are available to customers. Be sure to include the services that the Postal Service agrees are available to you.'),
   );
 
-  $form['international']['uc_usps_intl_services'] = array(
+  $form['international']['legacy']['uc_usps_intl_services'] = array(
     '#type' => 'checkboxes',
     '#title' => t('USPS international parcel services'),
-    '#default_value' => $config->get('uc_usps_intl_services'),
     '#options' => _uc_usps_intl_services(),
+    '#default_value' => $config->get('uc_usps_intl_services'),
+    '#description' => t('Select the USPS services that are available to customers. Be sure to include the services that the Postal Service agrees are available to you.'),
+  );
+  $form['international']['newapi'] = array(
+    '#type' => 'fieldset',
+    '#states' => array(
+      'visible' => array(
+        ':input[name="uc_usps_authorization_method"]' => array('value' => 'newapi'),
+      ),
+    ),
+  );
+  $form['international']['newapi']['uc_usps_newapi_intl_env_services'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('USPS international envelope services'),
+    '#options' => $config->get('uc_usps_newapi_intl_env_all_services'),
```

Comment:
Does the USPS API provide the list of options? If there are new options, would you add an update hook to add them to the module?

Reply:
USPS does not provide a list of services outside of simply doing lookups to find them. They also do not provide unique keys for each option: there are SKUs and rate codes, but neither is unique (i.e., multiple logically distinct services can have the same SKU and/or rate code). There is a function in the .devel file that creates the list of options for each combination (and these are now provided as functions, not config data, similarly to the legacy services lists).